src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5855,6 +5867,233 @@ private BoundExpression BindTypeParameterCreationExpression(ObjectCreationExpres
             return MakeBadExpressionForObjectCreation(node, typeParameter, analyzedArguments, initializerOpt, typeSyntax, diagnostics);
         }
 
+#nullable enable
+        private BoundExpression BindCollectionLiteralExpression(CollectionCreationExpressionSyntax syntax, TypeSymbol targetType, BindingDiagnosticBag diagnostics)
+        {
+            // Follows approach from BindCollectionInitializerExpression.
+
+            // PROTOTYPE: Test type parameter as collection type.
+            MethodSymbol? constructor = null;
+            bool useAdd = false;
+            if (targetType is NamedTypeSymbol namedType)
+            {
+                // PROTOTYPE: Test constructor with all optional parameters; with params parameter.
+                constructor = namedType.Constructors.SingleOrDefault(c => c.ParameterCount == 0);
```

Comment:
nit: SingleOrDefault will throw if htere are multiple.  which coudl happen in error recovery scenarios i believe.